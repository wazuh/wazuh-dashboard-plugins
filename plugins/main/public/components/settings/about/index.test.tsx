import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SettingsAbout } from '.';

jest.mock('./appInfo', () => ({
  SettingsAboutAppInfo: jest.fn().mockReturnValue(<div>App info</div>),
}));

jest.mock('./generalInfo', () => ({
  SettingsAboutGeneralInfo: jest.fn().mockReturnValue(<div>General info</div>),
}));

describe('SettingsAbout component', () => {
  test('should render a component with a loader', () => {
    const { container } = render(<SettingsAbout pluginAppName="Wazuh dashboard" />);

    expect(container).toMatchSnapshot();

    const loaders = container.getElementsByClassName('euiLoadingContent');
    expect(loaders.length).toBe(1);
  });

  test('should render a component without a loader', () => {
    const { container, getByText } = render(
      <SettingsAbout
        appInfo={{
          'app-version': '4.8.0',
          revision: '01',
          installationDate: 'Sep 25, 2023 @ 14:03:40.816',
        }}
        pluginAppName="Wazuh dashboard"
      />
    );

    expect(container).toMatchSnapshot();

    const loaders = container.getElementsByClassName('euiLoadingContent');
    expect(loaders.length).toBe(0);

    const appInfo = getByText('App info');
    expect(appInfo).toBeInTheDocument();
  });
});
