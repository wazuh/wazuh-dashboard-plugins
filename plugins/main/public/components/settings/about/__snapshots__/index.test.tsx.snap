// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsAbout component should render a component with a loader 1`] = `
<div>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <div>
        General info
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <span
          class="euiLoadingContent"
        >
          <span
            class="euiLoadingContent__singleLine"
          >
            <span
              class="euiLoadingContent__singleLineBackground"
            />
          </span>
          <span
            class="euiLoadingContent__singleLine"
          >
            <span
              class="euiLoadingContent__singleLineBackground"
            />
          </span>
          <span
            class="euiLoadingContent__singleLine"
          >
            <span
              class="euiLoadingContent__singleLineBackground"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SettingsAbout component should render a component without a loader 1`] = `
<div>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <div>
        General info
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <div>
          App info
        </div>
      </div>
    </div>
  </div>
</div>
`;
